src/Core/Framework/DependencyInjection/cache.xml

```diff
@@ -195,7 +188,7 @@
 
         <service id="shopware.reverse_proxy.http_client" class="GuzzleHttp\Client"/>
 
-        <service id="Shopware\Core\Framework\Adapter\Cache\ReverseProxy\VarnishReverseProxyGateway">
+        <service id="Shopware\Core\Framework\Adapter\Cache\ReverseProxy\AbstractReverseProxyGateway" class="Shopware\Core\Framework\Adapter\Cache\ReverseProxy\VarnishReverseProxyGateway">
```

Comment:
this is now the default definition, before it was removed with the cache rework where we removed the redis + varnish combination implementation, however now the declaration of the the abstract class was missing, that wen't unnoticed, because the compiler pass will remove the services if you don't actively enable reverse proxy

cc @mitelg & @shyim